cmd/istioctl/main.go

```diff
@@ -223,3 +247,64 @@ func readInput() (proto.Message, error) {
 
 	return v, nil
 }
+
+// readInput reads multiple documents from the input and checks with the schema
+func readInputs() ([]MetadataAndRule, error) {
+
+	var reader io.Reader
+	var err error
+
+	if file == "" {
+		reader = os.Stdin
+	} else {
+		reader, err = os.Open(file)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var varr []MetadataAndRule
+
+	// We store route-rules as a YaML stream; there may be more than one decoder.
+	var yamlDecoder *yaml.YAMLOrJSONDecoder = yaml.NewYAMLOrJSONDecoder(reader, 512*1024)
```

Comment:
You can use [https://golang.org/ref/spec#Short_variable_declarations](https://golang.org/ref/spec#Short_variable_declarations) to simplify the code and avoid redeclaring the yamlDecode type, e.g.

```
yamlDecoder := yaml.NewYAMLOrJSONDecoder(reader, 512*1024)
```